lib/valueflow.cpp

```diff
@@ -444,6 +444,18 @@ static void setTokenValue(Token* tok, const ValueFlow::Value &value, const Setti
             return;
         }
 
+        // known result when a operand is true.
+        if (Token::simpleMatch(parent, "&&") && value.isKnown() && value.isIntValue() && value.intvalue==0) {
```

Comment:
I see no test case for this

Reply:
you don't need to check if the value is "known".

Reply:
Actually, this shouldn't be in the PR, I'll remove it. This is part of PR #1294, which does have test cases.